connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -432,7 +439,9 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
             final SchemaAndValue value;
             try {
-                value = converter.toConnectData(topic, record.value());
+                synchronized (lock) {
```

Comment:
Do we need `Converter` access to be synchronized? I think it is reasonable to expect them to be thread-safe.

Nit: having this inline this seems inconsistent with having a `convertFromConnectData` method.